### PR TITLE
Task16 implement home page

### PR DESCRIPTION
**Pull request by Platton96**

### CODE REVIEW

File: MoneySplitter.Models/App/SortParameter.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace MoneySplitter.Models.App
+{
+    public enum SortParameter
+    {
+        CREATING_DATE,
+        TRANSACTION_DATE,
+        USER_ROLE,
+        SINGLE_COST,
+        NAME_TRANSACTION
```
Comment (Vovanella95) [inline MoneySplitter.Models/App/SortParameter.cs]: TRANSACTION_NAME
CREATION_DATE

File: MoneySplitter.Models/TransactionEventModel.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace MoneySplitter.Models
+{
+    public class TransactionEventModel
+    {
+		public TransactionModel RootTransaction { get; set; }
```
Comment (Vovanella95) [inline MoneySplitter.Models/TransactionEventModel.cs]: correct alignment

File: MoneySplitter.Models/UserRole.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@ public enum UserRole
         IN_BEGIN,
         IN_PROGRESS,
         FINISHED,
+        USER_TRANSACTION,
```
Comment (Vovanella95) [inline MoneySplitter.Models/UserRole.cs]: Not informational, maybe better rename to just MY or OWNER

File: MoneySplitter.Win10/Common/NavigateManager.cs [hunk 4]
```diff
@@ -9,91 +10,101 @@
...
+		#region Fields and constructor
+		private readonly INavigationService _windowNavigationService;
+		private INavigationService _shellNavigationService;
+
+		public event EventHandler<Type> OnShellNavigationManagerNavigated;
+
+		public NavigationManager(INavigationService navigationService)
+		{
+			_windowNavigationService = navigationService;
+
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/NavigateManager.cs]: Empty line

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+		public IEnumerable<TransactionEventModel> GetTransactionEvents(IEnumerable<TransactionModel> friendTransactions,
+			int? friendId = null,
+			bool isForceHide = false,
+			bool isDeadLineShowed = false)
+		{
+			return friendTransactions
+                .Select(tr => ConvertTransactionModelToTransactionEventModel(tr, friendId, isForceHide, isDeadLineShowed));
+		}
+        public IEnumerable<TransactionEventModel> GetFriendDebts(IEnumerable<TransactionModel> friendTransactions, int friendId)
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: no space

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        {
+            return GetTransactionEvents(friendTransactions.
+                Where(tr => tr.Owner.Id == _membershipService.CurrentUser.Id), friendId, false, true);
+     
+        }
+
+        public IEnumerable<TransactionEventModel> GetFriendLends(IEnumerable<TransactionModel> friendTransactions, int friendId)
+        {
+
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: check spaces everywhere

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+		private int GetNotVisibilCollabarratorsCount(TransactionModel transactionModel)
+		{
+			var VisibilUserWithoutMeAndOwnerCount = transactionModel.Owner.Id == _membershipService.CurrentUser.Id ? 2 : 1;
+
+			return transactionModel.Collaborators.Count(cl => cl.Id != transactionModel.Owner.Id &&
+															  cl.Id != _membershipService.CurrentUser.Id) - VisibilUserWithoutMeAndOwnerCount;
+
+		}
+
+		private IEnumerable<string> GetVicibilCollaboratorsImageUrls(TransactionModel transactionModel)
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: What? ))))) GetVisibleCollaboratorsImageUrls

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+		{
+			const int MAX_COUNT_IMAGE = 4;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: const must not be in method

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+			var OwnerAndMe = new List<string>();
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: camelCase + strange name

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+			OwnerAndMe.Add(transactionModel.Owner.ImageUrl);
+
+			if (transactionModel.Owner.Id != _membershipService.CurrentUser.Id)
+			{
+				OwnerAndMe.Add(_membershipService.CurrentUser.ImageUrl);
+			}
+
+			var allCollaborators = OwnerAndMe.Concat(transactionModel.Collaborators.
+														Where(cl => cl.Id != transactionModel.Owner.Id && cl.Id != _membershipService.CurrentUser.Id)
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: omg what a spaces

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+														.Select(cl => cl.ImageUrl));
+
+			return allCollaborators.Count() <= MAX_COUNT_IMAGE ? allCollaborators : allCollaborators.Take(MAX_COUNT_IMAGE - 1);
+		}
+
+		private TypeDate GetTypeDate(TransactionModel transactionModel, bool IsDeadLineShowed)
+		{
+			if (IsDeadLineShowed ||
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: symbols should be moved to second line,  I mean:
condition1
|| condition2
&& condition3....

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+				transactionModel.OngoingDate == null ||
+				transactionModel.DeadlineDate < transactionModel.OngoingDate)
+			{
+				return TypeDate.DEADLINE_DATE;
+			}
+
+			return TypeDate.ONGOIND_DATE;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: DateType

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+			if (typeDate == TypeDate.DEADLINE_DATE)
+			{
+				return transactionModel.DeadlineDate;
+			}
+
+			return (DateTime)transactionModel.OngoingDate;
+		}
+
+		private TransactionEventModel ConvertTransactionModelToTransactionEventModel(TransactionModel transactionModel,
+			int? friendId,
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: if you movind parameters to new lines, first parameter must be moved too
MethodName(
parameter1,
parameter2,
...)

File: MoneySplitter.Win10/Converters/BankCardConverter.cs [hunk 14]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using Windows.UI.Xaml.Data;
+
+namespace MoneySplitter.Win10.Converters
+{
+    public class BankCardConverter : IValueConverter
+    {
+        object IValueConverter.Convert(object value, Type targetType, object parameter, string language)
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Converters/BankCardConverter.cs]: better not use Interface definer here, only Convert(...

File: MoneySplitter.Win10/CustomControls/TransactionEventControl.xaml.cs [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace MoneySplitter.Win10.CustomControls
+{
+	public sealed partial class TransactionEventControl : UserControl
+	{
+		public event EventHandler<TransactionEventModel> OnGiveTransactionButtonClick;
+		public event EventHandler<TransactionEventModel> OnRemindTransactionButtonClick;
+		public event EventHandler<TransactionEventModel> OnApproveTransactionButtonClick;
+
+
```
Comment (Vovanella95) [inline MoneySplitter.Win10/CustomControls/TransactionEventControl.xaml.cs]: empty spaaaaaaaaaaaaces
what are we living for....

File: MoneySplitter.Win10/Views/FriendDetailsView.xaml.cs [hunk 16]
```diff
@@ -0,0 +1,28 @@
+﻿using MoneySplitter.Win10.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Views/FriendDetailsView.xaml.cs]: too many references

File: MoneySplitter.Win10/Views/SearchUsersView.xaml.cs [hunk 17]
```diff
@@ -31,5 +31,10 @@ private void OnTextBoxTextChanged(object sender, TextChangedEventArgs e)
         {
             await ViewModel.AddFriendAsync(e.Id);
         }
+
+        private void UsersGridView_ItemClick(object sender, ItemClickEventArgs e)
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Views/SearchUsersView.xaml.cs]: is it needed?

File: MoneySplitter.Models/App/SortParameter.cs (hunk 1 above)
Comment (Platton96) [reply MoneySplitter.Models/App/SortParameter.cs]: rename

File: MoneySplitter.Models/TransactionEventModel.cs (hunk 2 above)
Comment (Platton96) [reply MoneySplitter.Models/TransactionEventModel.cs]: fixed

File: MoneySplitter.Models/UserRole.cs (hunk 3 above)
Comment (Platton96) [reply MoneySplitter.Models/UserRole.cs]: rename

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs (hunk 5 above)
Comment (Platton96) [reply MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: remove space

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs (hunk 6 above)
Comment (Platton96) [reply MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: cheak

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs (hunk 7 above)
Comment (Platton96) [reply MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: rename

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs (hunk 9 above)
Comment (Platton96) [reply MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: fixed

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs (hunk 11 above)
Comment (Platton96) [reply MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: fixed

File: MoneySplitter.Win10/Common/TransactionEventModelFactory.cs (hunk 13 above)
Comment (Platton96) [reply MoneySplitter.Win10/Common/TransactionEventModelFactory.cs]: fixed

File: MoneySplitter.Win10/CustomControls/TransactionEventControl.xaml.cs (hunk 15 above)
Comment (Platton96) [reply MoneySplitter.Win10/CustomControls/TransactionEventControl.xaml.cs]: delete

File: MoneySplitter.Win10/Views/SearchUsersView.xaml.cs (hunk 17 above)
Comment (Platton96) [reply MoneySplitter.Win10/Views/SearchUsersView.xaml.cs]: it is for future but i remove it

File: MoneySplitter.Win10/Views/FriendDetailsView.xaml.cs (hunk 16 above)
Comment (Platton96) [reply MoneySplitter.Win10/Views/FriendDetailsView.xaml.cs]: remove

File: MoneySplitter.Win10/Converters/BankCardConverter.cs (hunk 14 above)
Comment (Platton96) [reply MoneySplitter.Win10/Converters/BankCardConverter.cs]: fixed

**Pull request merged**